pkg/cmd/pulumi/about.go

```diff
@@ -0,0 +1,304 @@
+				" - which OS pulumi was run from\n" +
+				" - the current project\n" +
+				" - the current stack\n" +
+				" - the current backend\n",
+			Args: cmdutil.MaximumNArgs(0),
+			Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+				var err error
+				opts := display.Options{
+					Color: cmdutil.GetGlobalColorization(),
+				}
+
+				// TODO Maybe include a message saying to include this info when
+				// starting an issue.
+
+				formatCLIAbout()
+				fmt.Print("\n")
+
+				if err = formatPluginAbout(); err != nil {
+					err = errors.Wrap(err, "Failed to get information about the plugin.")
+					warn(err, &opts)
+				}
+				fmt.Print("\n")
+
+				if err = formatHostAbout(); err != nil {
+					err = errors.Wrap(err, "Failed to get information about the host")
+					warn(err, &opts)
+				}
+				fmt.Print("\n")
+
+				var proj *workspace.Project
+				var pwd string
+				if proj, pwd, err = readProject(); err != nil {
+					err = errors.Wrap(err, "Failed to read project for diagnosis")
+					warn(err, &opts)
+				} else {
+					fmt.Printf("This is a %s project.\n\n", proj.Runtime.Name())
+
+					if err = formatProgramDependenciesAbout(proj.Runtime.Name(), pwd); err != nil {
+						err = errors.Wrap(err, "Failed to get diagnositc information about the puluimi program's plugins")
+						warn(err, &opts)
+					}
+					fmt.Print("\n")
+				}
+
+				var backend backend.Backend
+				backend, err = currentBackend(display.Options{Color: cmdutil.GetGlobalColorization()})
+				if err != nil {
+					err = errors.Wrapf(err, "Could not access the backend to give diagnosis.")
+					warn(err, &opts)
+				} else {
+					if err = formatCurrentStackAbout(backend); err != nil {
+						err = errors.Wrap(err, "Failed to get diagnostic information about the current stack")
+						warn(err, &opts)
+					}
+					fmt.Print("\n")
+
+					if err = formatBackendAbout(backend); err != nil {
+						errors.Wrap(err, "Failed to gather diagnostic information on the current backend")
+						warn(err, &opts)
+					}
+
+				}
+
+				return nil
+			},
+			),
+		}
+
+	return cmd
+}
+
+func formatPluginAbout() error {
+	var plugins []workspace.PluginInfo
+	var err error
+	plugins, err = workspace.GetPluginsWithMetadata()
+	if err != nil {
+		return err
+	}
+	fmt.Print("Plugins\n")
+	sort.Slice(plugins, func(i, j int) bool {
+		pi, pj := plugins[i], plugins[j]
+		if pi.Name < pj.Name {
+			return true
+		} else if pi.Name == pj.Name && pi.Kind == pj.Kind &&
+			(pi.Version == nil || (pj.Version != nil && pi.Version.GT(*pj.Version))) {
+			return true
+		}
+		return false
+	})
+	rows := []cmdutil.TableRow{}
+	for _, plugin := range plugins {
+		name := plugin.Name
+		version := plugin.Version.String()
+		rows = append(rows, cmdutil.TableRow{
+			Columns: []string{name, version},
+		})
+	}
+	cmdutil.PrintTable(cmdutil.Table{
+		Headers: []string{"NAME", "VERSION"},
+		Rows:    rows,
+	})
+	return nil
+}
+
+func formatHostAbout() error {
```

Comment:
We don't have lots of test coverage for the CLI - let's take this opportunity to get some tests added :D